### PR TITLE
License Info is Displayed as Clickable Link in Gem Catalog + Other Inspector Improvements

### PR DESCRIPTION
**Pull request by AMZN-nggieber**
Also some general improvements to gem inspector including: Adding text eliding, fixing typos, depending gems is a conditional element now and conditional spacing.
Additionally adds license and license_url as modifiable properties in in o3de scripts.

HQX9QQIwT1
eja0Yt75d0

F:\o3de\o3de\python>python.cmd -m pytest F:\o3de\o3de\scripts\o3de\tests\unit_test_gem_properties.py --build-directory F:\o3de\o3de\build\windows_vs2019\bin\profile
================================================= test session starts =================================================
platform win32 -- Python 3.7.10, pytest-5.3.2, py-1.9.0, pluggy-0.13.1
rootdir: F:\o3de\o3de, inifile: pytest.ini
plugins: mock-2.0.0, timeout-1.3.4, ly-test-tools-1.0.0
collected 3 items

..\scripts\o3de\tests\unit_test_gem_properties.py ...                                                            [100%]

================================================== 3 passed in 0.05s ==================================================

### CODE REVIEW

**Comment by AMZN-alexpete** [comment]
This is great, but I think the license URL for all of our gems should point to LICENSE.txt in our GitHub repo https://github.com/o3de/o3de/blob/development/LICENSE.txt which covers MIT, Apache and other specifics about the licenses, and will also be kept up to date incase the license changes.

File: Code/Tools/ProjectManager/Source/GemCatalog/GemInspector.cpp [hunk 1]
```diff
@@ -52,40 +52,60 @@ namespace O3DE::ProjectManager
...
-        m_dependingGems->Update("Depending Gems", "The following Gems will be automatically enabled with this Gem.", m_model->GetDependingGemNames(modelIndex));
+        QStringList dependingGems = m_model->GetDependingGemNames(modelIndex);
+        if (!dependingGems.isEmpty())
+        {
+            m_dependingGems->Update("Depending Gems", "The following Gems will be automatically enabled with this Gem.", dependingGems);
+            m_dependingGems->show();
+        }
+        else
+        {
+            m_dependingGems->hide();
```
Comment (AMZN-alexpete) [inline Code/Tools/ProjectManager/Source/GemCatalog/GemInspector.cpp:107]: excellent!

**Review by AMZN-ScottR — APPROVED** [review]
Could use a `tr()` pass on display strings but looks fine otherwise.

File: Templates/AssetGem/Template/gem.json [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {
     "gem_name": "${Name}",
     "display_name": "${Name}",
-    "license": "What license ${Name} uses goes here: i.e. https://opensource.org/licenses/MIT",
+    "license": "What license ${Name} uses goes here: i.e. Apache-2.0 Or MIT",
+    "license_url": "https://github.com/o3de/o3de/blob/development/LICENSE.txt",
```
Comment (AMZN-alexpete) [inline Templates/AssetGem/Template/gem.json:5]: let's not make the default license URL point to our repo

File: Templates/CppToolGem/Template/gem.json [hunk 3]
```diff
@@ -1,7 +1,8 @@
 {
     "gem_name": "${Name}",
     "display_name": "${Name}",
-    "license": "What license ${Name} uses goes here: i.e. https://opensource.org/licenses/MIT",
+    "license": "What license ${Name} uses goes here: i.e. Apache-2.0 Or MIT",
+    "license_url": "https://github.com/o3de/o3de/blob/development/LICENSE.txt",
```
Comment (AMZN-alexpete) [inline Templates/CppToolGem/Template/gem.json:5]: ditto here, this should be blank

File: Templates/DefaultGem/Template/gem.json [hunk 4]
```diff
@@ -1,7 +1,8 @@
 {
     "gem_name": "${Name}",
     "display_name": "${Name}",
-    "license": "What license ${Name} uses goes here: i.e. https://opensource.org/licenses/MIT",
+    "license": "What license ${Name} uses goes here: i.e. Apache-2.0 Or MIT",
+    "license_url": "https://github.com/o3de/o3de/blob/development/LICENSE.txt",
```
Comment (AMZN-alexpete) [inline Templates/DefaultGem/Template/gem.json:5]: ditto here, this should be blank

File: Templates/PythonToolGem/Template/gem.json [hunk 5]
```diff
@@ -1,7 +1,8 @@
 {
     "gem_name": "${Name}",
     "display_name": "${Name}",
-    "license": "What license ${Name} uses goes here: i.e. https://opensource.org/licenses/MIT",
+    "license": "What license ${Name} uses goes here: i.e. Apache-2.0 Or MIT",
+    "license_url": "https://github.com/o3de/o3de/blob/development/LICENSE.txt",
```
Comment (AMZN-alexpete) [inline Templates/PythonToolGem/Template/gem.json:5]: ditto here, this should be blank

**Review by AMZN-alexpete — CHANGES_REQUESTED** [review]
Looks good, just one change which is to not use the O3DE license as the default license_url for the templates.

**Review by AMZN-alexpete — APPROVED** [review]

**Review by AMZN-Phil — APPROVED** [review]

**Pull request merged**